sdk/log/provider_test.go

```diff
@@ -308,3 +308,18 @@ func BenchmarkLoggerProviderLogger(b *testing.B) {
 	b.StopTimer()
 	loggers[0].Enabled(context.Background(), log.Record{})
 }
+
+func BenchmarkLoggerProviderLoggerSame(b *testing.B) {
+	p := NewLoggerProvider()
+
+	b.ResetTimer()
+	b.ReportAllocs()
+
+	var logger log.Logger
+	for i := 0; i < b.N; i++ {
+		logger = p.Logger("test", log.WithInstrumentationVersion("v1.2.3"))
+	}
+
+	b.StopTimer()
+	logger.Enabled(context.Background(), log.Record{})
```

Comment:
Why is this done instead of something like `_ = logger`?

Reply:
Copied from previous benchmark. I think it OK as it has lower chances that it wouldn't be pruned by the compiler as it may see that it is a noop.